src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -63,14 +72,18 @@ public interface IHostEnvironment : IChannelProvider, IProgressChannelProvider
         IHost Register(string name, int? seed = null, bool? verbose = null);
 
         /// <summary>
-        /// Flag which indicate should we stop any code execution in this host.
+        /// The catalog of loadable components (<see cref="LoadableClassAttribute"/>) that are available in this host.
         /// </summary>
-        bool IsCancelled { get; }
+        ComponentCatalog ComponentCatalog { get; }
+    }
 
+    [BestFriend]
+    internal interface ICancelableHost
+    {
         /// <summary>
-        /// The catalog of loadable components (<see cref="LoadableClassAttribute"/>) that are available in this host.
+        /// Signal to stop exection in this host and all its children.
         /// </summary>
-        ComponentCatalog ComponentCatalog { get; }
+        void StopExecution();
```

Comment:
(minor) it's a bit of inconsistent naming - this is a "cancelable" thing, but it doesn't have a "Cancel" method - it has `StopExecution()`.